src/hub.html

```diff
@@ -605,20 +605,18 @@
                 vr-mode-toggle-visibility
                 vr-mode-toggle-playing__hud-controller
             >
-                <a-entity visibility-while-frozen="requireHoverOnNonMobile: false;" rotation="30 0 0">
-                    <a-entity class="ui interactable-ui">
-                        <a-entity class="leave-btn" position="0 -0.2 0" scale="0.7 0.7 0.7" leave-room-button is-remote-hover-target tags="singleActionButton: true;">
-                            <a-entity mixin="rounded-text-action-button" slice9="width: 0.8" scale="0.8 0.8 1.0">
-                                <a-entity text=" value:leave room; width:2; align:center;" text-raycast-hack position="0 0 0.01"></a-entity>
-                            </a-entity>
+                <a-entity class="ui interactable-ui" visibility-while-frozen="requireHoverOnNonMobile: false;" rotation="30 0 0">
```

Comment:
I think you liked the separation of these layers for architectural reasons, but they seem undesirable from a perf perspective. The outer three layers seem easy to combine into one. Combining those three with the inner-most entity will involve more work that doesn't seem important to do now.